### PR TITLE
fix missing anchor in external xref

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/UriUtility.cs [hunk 1]
```diff
@@ -78,5 +80,14 @@ public static string GetQueryStringAndFragment(string uriString)
...
             return index == -1 ? string.Empty : uriString.Substring(index);
         }
+
+        public static bool IsAbsolutePath(string uriString)
+        {
+            if (uriString == null)
+            {
+                throw new ArgumentNullException(nameof(uriString));
+            }
+            return AbsoluteUriRegex.Match(uriString).Success;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/UriUtility.cs]: >AbsoluteUriRegex [](start = 19, length = 16)

`/abc` should also be considered as absolute path?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Common/UriUtility.cs]: resolved

**Comment by vwxyzh** [comment]
#1655

File: test/Microsoft.DocAsCode.Common.Tests/UriUtilityTest.cs [hunk 2]
```diff
@@ -30,5 +30,27 @@ public void TestUriUtility(string input, string path, string queryString, string
             Assert.Equal(queryStringAndFragment, UriUtility.GetQueryStringAndFragment(input));
             Assert.Equal(nonFragment, UriUtility.GetNonFragment(input));
         }
+
+        public void TestIsAbsoluteUri()
```
Comment (hellosnow) [inline test/Microsoft.DocAsCode.Common.Tests/UriUtilityTest.cs]: >TestIsAbsoluteUri [](start = 20, length = 17)

[fact]?

File: test/Microsoft.DocAsCode.Common.Tests/UriUtilityTest.cs [hunk 3]
```diff
@@ -30,5 +30,27 @@ public void TestUriUtility(string input, string path, string queryString, string
...
+            Assert.False(UriUtility.IsAbsoluteUri("relative/path/to/resource.txt"));
+            Assert.False(UriUtility.IsAbsoluteUri("#frag01"));
+            Assert.False(UriUtility.IsAbsoluteUri(string.Empty));
+        }
+
+        public void TestIsAbsolutePath()
+        {
+            Assert.True(UriUtility.IsAbsolutePath("/catalog/shownew.htm"));
+
+            Assert.False(UriUtility.IsAbsoluteUri("http://www.contoso.com/"));
```
Comment (hellosnow) [inline test/Microsoft.DocAsCode.Common.Tests/UriUtilityTest.cs]: >False [](start = 19, length = 5)

for the same judgement, result should align with line 36?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/UriUtility.cs [hunk 4]
```diff
@@ -78,5 +81,24 @@ public static string GetQueryStringAndFragment(string uriString)
...
+        public static bool IsAbsoluteUri(string uriString)
+        {
+            if (uriString == null)
+            {
+                throw new ArgumentNullException(nameof(uriString));
+            }
+            return AbsoluteUriRegex.Match(uriString).Success;
+        }
+
+        public static bool IsAbsolutePath(string uriString)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/UriUtility.cs]: >IsAbsolutePath [](start = 27, length = 14)

Put into PathUtility?

**Pull request merged**